src/components/dialogs/CountryCatalogDialog.tsx

```diff
@@ -0,0 +1,118 @@
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  Divider,
+  Typography,
+  Button,
+  DialogTitle,
+  Dialog,
+  FormControl,
+  InputLabel,
+  Select,
+  MenuItem,
+  TextField,
+  TableContainer,
+  Paper,
+  CircularProgress,
+  Box,
+  Grid,
+} from "@mui/material";
+import ViewInArIcon from "@mui/icons-material/ViewInAr";
+import { Goods, ReportProps } from "./ReportedGoodsDialog.types";
+import { Field, FormikProvider, useFormik } from "formik";
+import { Type, Values } from "../forms/report-form/CreateReport.types";
+import * as yup from "yup";
+import { CountryData } from "./CountryCatalogDialog.types";
+import { Country, State, City } from "country-state-city";
+import { Link, useNavigate, useParams } from "react-router-dom";
+import AddIcon from "@mui/icons-material/Add";
+import useMount from "../../services/isMountedHook";
+
+const validationSchema = yup.object({
+  name: yup
+    .string()
+    .min(3, "Country name length should be of minimum 3 characters")
+    .required("Name is required"),
+});
+
+export default function CountryCatalogDialog() {
+  const [open, setOpen] = useState(false);
+  const jwt = useSelector((state: RootState) => state.user.user.jwt);
+  const dispatch = useDispatch();
+  const initialValues = {
+    name: "",
+  };
+
+  const formik = useFormik({
+    initialValues,
+    enableReinitialize: true,
+    validationSchema: validationSchema,
+    onSubmit: (data: CountryData, { resetForm }) => {
+      clientApi.country
+        .create(jwt, data)
+        .then((response) => {
+          dispatch(clearError());
+          setOpen(false);
+          window.location.reload();
+        })
+        .catch((error) => {
+          resetForm({});
+          window.scrollTo(0, 0);
+        });
+      console.log("data", data);
+    },
+  });
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const twinkleBlue = "#e9ecef";
+  const headStyle = {
+    backgroundColor: twinkleBlue,
+  };
+
+  return (
+    <>
+      <Button onClick={() => handleClickOpen()}>
+        <AddIcon />
+      </Button>
+      <Box sx={{ minWidth: 300 }}>
+        <Dialog fullWidth={true} open={open} onClose={() => setOpen(false)}>
+          <DialogTitle>Add country</DialogTitle>
+          <Divider />
+          <form onSubmit={formik.handleSubmit}>
+            <TextField
+              fullWidth
+              id="name"
+              name="name"
+              label="Country"
+              value={formik.values.name}
+              onChange={formik.handleChange}
+              error={formik.touched.name && Boolean(formik.errors.name)}
+              helperText={formik.touched.name && formik.errors.name}
+              sx={{ mb: 3 }}
+            />
+            <Box sx={{ display: "flex" }}>
+              <Button
+                variant="contained"
+                type="submit"
+                style={{ margin: "0 0 10px 0" }}
+              >
+                Submit
+              </Button>
```

Comment:
Dialog doesn't close, when click on submit btn
And it doesn't add country in country table, only in console
![image](https://user-images.githubusercontent.com/60444425/175514745-c806c9a5-543d-4764-bbeb-87fcf41c1839.png)